storybook/stories/components/AppModal.stories.mdx

```diff
@@ -0,0 +1,49 @@
+import { Canvas, Meta, Story, ArgsTable, Source } from '@storybook/addon-docs';
+import AppModal from '@shell/components/AppModal';
+
+<Meta 
+  title="Components/AppModal" 
+  component={AppModal}
+/>
+
+export const Template = (args, { argTypes }) => ({
+  components: { AppModal },
+  props:      Object.keys(argTypes),
+  template:   '<app-modal v-bind="$props">Modal content goes here</app-modal>',
+})
+
+# Alert
```

Comment:
This should be `Modal`. 

Also we don't want to have specification code wise, rather how does it work and how it looks like the final outcome.
If we are planning to review the design, we can consider this in a second instance. However the final goal for Storybook was to create a case where we could actually trigger the modal.

Since we want to review the UI and the UX, we may think about this in a second iteration and not block the migration.

![Screenshot 2024-03-07 at 17 30 10](https://github.com/rancher/dashboard/assets/5009481/f6a1cfbc-3661-4c89-bd3e-b5b47d610aca)

Reply:
> This should be Modal.

gah, you caught my copy/paste job for templating this story. Thanks, good catch.

> Also we don't want to have specification code wise, rather how does it work and how it looks like the final outcome.
If we are planning to review the design, we can consider this in a second instance. However the final goal for Storybook was to create a case where we could actually trigger the modal.

That sounds good to me. We might require some additional research to see what a good example would look like for actually triggering a modal for Storybook.